avocado/core/job.py

```diff
@@ -474,6 +477,9 @@ def _run(self):
             stacktrace.log_exc_info(sys.exc_info(), 'avocado.app.debug')
             self._remove_job_results()
             raise exceptions.OptionValidationError(details)
+
+        self.job_pre_post_dispatcher.map_methods('pre', self)
```

Comment:
OK I'm sorry, you do run the post-hook outside this class in the `finally` block, so it is actually executed at any circumstances. I like that.